### PR TITLE
Run recreate index through a celery task

### PR DESCRIPTION
**Pull request by arslanashraf7**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
#4959 

#### What's this PR do?
- It moves the execution of `recreate_index` out of the main thread and puts it to a separate celery task

#### How should this be manually tested?
- Running the command e.g.  `./manage.py recreate_index` should now run separately in a celery task.

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
Previously we were facing timeouts when running this command with high amount of records.

### CODE REVIEW

**Comment by pdpinch** [comment]
Arslan, I know there are still some open questions about the issue this addresses, but you should still put in some minimal details in the PR description.

**Comment by arslanashraf7** [comment]
> Arslan, I know there are still some open questions about the issue this addresses, but you should still put in some minimal details in the PR description.

Thanks for pointing this out, I've added the initial details.

File: search/tasks.py [hunk 1]
```diff
@@ -136,3 +137,11 @@ def populate_query_memberships(percolate_query_id):
         percolate_query_id (int): Database id for the PercolateQuery to populate
     """
     api.populate_query_memberships(percolate_query_id)
+
+
+@app.task
```
Comment (rhysyngsun) [inline search/tasks.py:142]: So we can ensure this task doesn't get dropped.

```suggestion
@app.task(acks_late=True)
```

File: search/management/commands/recreate_index.py [hunk 2]
```diff
@@ -41,10 +41,10 @@ def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
             import uuid
             profile = cProfile.Profile()
             profile.enable()
-            recreate_index()
+            recreate_index_async.delay()
```
Comment (rhysyngsun) [inline search/management/commands/recreate_index.py:44]: A couple things:

- remove the profiling code, it's not doing anything anymore
- print out the task id so if it doesn't finish we can investigate

File: search/management/commands/recreate_index.py [hunk 3]
```diff
@@ -1,50 +1,19 @@
...
-            profile = cProfile.Profile()
-            profile.enable()
-            recreate_index()
-            profile.disable()
-            filename = 'recreate_index_{}.profile'.format(uuid.uuid4())
-            profile.dump_stats(filename)
-            self.stdout.write('Output profiling data to: {}'.format(filename))
-        else:
-            recreate_index()
+        recreate_index_async.delay()
```
Comment (noisecapella) [inline search/management/commands/recreate_index.py:19]: Can we write out the task id to the console here?

**Review by noisecapella — CHANGES_REQUESTED** [review]

File: search/management/commands/recreate_index.py (hunk 3 above)
Comment (arslanashraf7) [reply search/management/commands/recreate_index.py]: @noisecapella, Thanks for pointing this out I was writing the task id within the task but the command will be a better place for it. I've now updated the PR as per your suggestion and this is ready for review.

Also, Just FYI, There is another [PR](https://github.com/mitodl/micromasters/pull/4968) that is actually rebased on this one. That PR is the successor part of what we wanted to achieve in the ticket. The associated ticket has more details regarding that. You can take a look at that as well and we can decide either to merge this PR and check out the feasibility of the other one or close this PR in favor of the other [PR](https://github.com/mitodl/micromasters/pull/4968) which is rebased on this.

**Review by noisecapella — APPROVED** [review]

File: search/management/commands/recreate_index.py (hunk 3 above)
Comment (noisecapella) [reply search/management/commands/recreate_index.py]: I think it makes sense to merge this and then continue the work in #4968

**Pull request merged**